frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/create-vm-wizard.tsx

```diff
@@ -89,6 +94,16 @@ const kubevirtInterOP = async ({
   const clonedNetworks = _.cloneDeep(networks);
   const clonedStorages = _.cloneDeep(storages);
 
+  const cloudInitVolume = clonedStorages.map((stor) => stor.volume).find(getVolumeCloudInitNoCloud);
+  const data = VolumeWrapper.initialize(cloudInitVolume).getCloudInitNoCloud();
+  const cloudInitHelper = new CloudInitDataHelper(data);
+
+  if (
+    clonedVMsettings.hostname !== { value: cloudInitHelper.get(CloudInitDataFormKeys.HOSTNAME) }
```

Comment:
how could this work? This will return false every time.

We should just check if cloudinit hostname is not empty

Reply:
vmSettings.hostname is assigned a value here
https://github.com/openshift/console/pull/3400/files#diff-50461f01355ce217b3e262d34df0c900R122

if the user won't change the hostname in the cloudinit tab then they are equal. I guess the cloud init's host name is always the most updated one and we can just assign it twice for prettier code

Reply:
suppose there is no cloud init, you want to copy the template hostname regardless

please check, comparison of `anything !== {}` returns true  every time